Source/BoundaryConditions/PML.cpp

```diff
@@ -1139,41 +1196,87 @@ PushPMLPSATDSinglePatch (
     const int lev,
     SpectralSolver& solver,
     std::array<std::unique_ptr<amrex::MultiFab>,3>& pml_E,
-    std::array<std::unique_ptr<amrex::MultiFab>,3>& pml_B ) {
+    std::array<std::unique_ptr<amrex::MultiFab>,3>& pml_B,
+    std::unique_ptr<amrex::MultiFab>& pml_F,
+    std::unique_ptr<amrex::MultiFab>& pml_G) {
 
     using SpIdx = SpectralPMLIndex;
 
-    // Perform forward Fourier transform
-    // Note: the correspondance between the spectral PML index
-    // (Exy, Ezx, etc.) and the component (PMLComp::xy, PMComp::zx, etc.)
-    // of the MultiFabs (e.g. pml_E) is dictated by the
-    // function that damps the PML
+    // Perform forward Fourier transforms
     solver.ForwardTransform(lev, *pml_E[0], SpIdx::Exy, PMLComp::xy);
     solver.ForwardTransform(lev, *pml_E[0], SpIdx::Exz, PMLComp::xz);
-    solver.ForwardTransform(lev, *pml_E[1], SpIdx::Eyz, PMLComp::yz);
     solver.ForwardTransform(lev, *pml_E[1], SpIdx::Eyx, PMLComp::yx);
+    solver.ForwardTransform(lev, *pml_E[1], SpIdx::Eyz, PMLComp::yz);
     solver.ForwardTransform(lev, *pml_E[2], SpIdx::Ezx, PMLComp::zx);
     solver.ForwardTransform(lev, *pml_E[2], SpIdx::Ezy, PMLComp::zy);
     solver.ForwardTransform(lev, *pml_B[0], SpIdx::Bxy, PMLComp::xy);
     solver.ForwardTransform(lev, *pml_B[0], SpIdx::Bxz, PMLComp::xz);
-    solver.ForwardTransform(lev, *pml_B[1], SpIdx::Byz, PMLComp::yz);
     solver.ForwardTransform(lev, *pml_B[1], SpIdx::Byx, PMLComp::yx);
+    solver.ForwardTransform(lev, *pml_B[1], SpIdx::Byz, PMLComp::yz);
     solver.ForwardTransform(lev, *pml_B[2], SpIdx::Bzx, PMLComp::zx);
     solver.ForwardTransform(lev, *pml_B[2], SpIdx::Bzy, PMLComp::zy);
+
+    // WarpX::do_pml_dive_cleaning = true
+    if (pml_F)
```

Comment:
here you test `if(pml_F)` instead of `if(m_dive_cleaning)` to silence a compiler warning concerning the use of possibly uninitialized `pml_F`, right?

Reply:
I think I did it like this just because it was done in a similar way somewhere else in the code, but your explanation makes sense. I mean, what you say is probably why it was done like this somewhere else in the code. I have not checked it specifically, though.